src/EditorFeatures/Core/Implementation/IntelliSense/EditorCompletionImplementation/CompletionCommitManager.cs

```diff
@@ -66,6 +71,46 @@ public EditorCompletion.CommitResult TryCommit(ITextView view, ITextBuffer buffe
             return new EditorCompletion.CommitResult(isHandled: false, EditorCompletion.CommitBehavior.None);
         }
 
+        /// <summary>
+        /// This method needs to support custom procesing of commit characters to be on par with the old completion implementation.
+        /// TODO we should consider removign this method and arrange better processing of commit characters.
+        /// </summary>
+        private  bool IsCommitCharacter(char typeChar, ImmutableArray<CharacterSetModificationRule> rules)
+        {
+            // Tab, Enter and Null (call invoke commit) are always a commit character
+            if (typeChar == '\t' || typeChar == '\n' || typeChar == '\0') 
```

Comment:
it seriously worries me that:

1. Enter is being sent through as typing '\n'.
2. Commit is being sent through as typing \0

Reply:
Thank you, Cyrus!
I just tried to be on par (comparing unit tests) with the old implementation. However, as I understand, previously we were able to handle e.g. Enter independently on all other key strokes. Now we have to make almost all decisions within the TryCommit method. So, we should process '\n' and '0' here. Do you have a suggestion?

---
In reply to: [200802172](https://github.com/dotnet/roslyn/pull/28219#discussion_r200802172) [](ancestors = 200802172)